adagrams/game.py

```diff
@@ -1,11 +1,105 @@
+import random
+
+
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+    drawn_letters = [] 
+    for letter in range(10):
+        my_current_key = random.choice(list(LETTER_POOL))
+        my_current_value = LETTER_POOL.get(my_current_key)
+        if my_current_value != 0:
+            drawn_letters.append(my_current_key)
+            LETTER_POOL[my_current_key] = my_current_value -1
+        if LETTER_POOL[my_current_key] == 0:
+            LETTER_POOL.pop(my_current_key)
+    return drawn_letters
+            
+       
+def uses_available_letters(word, letter_bank): 
+    #check parameter if characters exist in  letter_bank
+    # true f all letters exist
+    # false if any letter is missing
+    # word is a string
+    #letter_bank a list of ten characters
+
+    bank_copy = letter_bank.copy()
```

Comment:
Great use of the copy function here! We definitely want to make sure our dictionary is copied over as opposed to manipulating the original dictionary! For future reference, feel free to take a look at the difference between .copy and .deepcopy. .copy will copy over references to items within the list or dictionary rather than make a copy of the lists within. This won't cause an issue for this project, but later on when working with nested data structures, you could run into problems!

https://realpython.com/copying-python-objects/